### ISSUE TITLE
Bug: Unable to enter maximum amount for NumberInputs

### ISSUE THREAD

**Issue by mobiusTripper-crypto** [decent-dao/fractal-interface#754]
We are using NumberInput which maxes out at Javascript `number` type. Change to Input and limit to ethers --- constants.MaxInt256 (<- I think)

![Image](https://user-images.githubusercontent.com/99622829/206479405-fa1745d2-4cb1-47f3-a82b-9d23139fa949.png)

**Comment by herbig** [comment]
@mobiusTripper-crypto @Da-Colon this is something we should probably fix before launch

**Comment by sethhrbek** [comment]
@herbig @mobiusTripper-crypto @Da-Colon  -- I think the scope for this task should be as below. Leave feedback if you feel differently

1. Change from `NumberInput` to use normal string Input. We can use `NumberInput` as a guide for the new component.
2. This new input should:
a. Handle Decimals -- if the user doesn't enter a decimal, then the number entered is considered whole units (1 ETH, 2 ETH, etc)
3. Input should have validation in place to not allow more than "x" characters to the right of the decimal point, where "x" is the number of decimals in the ERC20
4. Don't allow negative numbers
5. Add min/max if necessary

**Nice to have:**
Get the ERC20 symbol (ETH, FRAC, etc) and include that as a "rightIcon" on the input (or whatever chakra calls it)

If we need this input to handle integers, make `hasDecimals` property. If this property is false, then we assume this input is an integer, and can parse like so.

---

Once this is done, we can either, in a different task:
1. Replace the relevant inputs in the app with this new component locally
2. Abstract this component to the Design system and import it from there to use in the app

**Comment by mobiusTripper-crypto** [comment]
Upon further evaluation, I do not think min / max on the input is a good idea, as the value changes without any feedback to the user if we just change the value. I think using validation is a more appropriate solution to insure min / max values.
